bin/make_charts.py

```diff
@@ -0,0 +1,363 @@
+        )
+
+        quantiles(
+            output, 
+            [
+                Measures(values=data["aggregatedPublishLatencyQuantiles"], log="p-p.log", name="publish p-values"),
+                Measures(values=data["aggregatedEndToEndLatencyQuantiles"], log="p-e2e.log", name="end-to-end p-values")
+            ],
+            "quantiles",
+            prefix+"quantiles"
+        )
+    else:
+        single(
+            output,
+            Measures(values=data["sent"], log="sent.log", name="sent"),
+            dt=dt,
+            output_name="throughput",
+            title=prefix + f"throughput (ops, {int(dt)}s agg)"
+        )
+
+        quantiles(
+            output, 
+            [
+                Measures(values=data["aggregatedPublishLatencyQuantiles"], log="p-p.log", name="publish p-values")
+            ],
+            "quantiles",
+            prefix+"quantiles"
+        )
+
+def bar(output, datasets, output_name, title):
+    datasets.sort(key=lambda x:x["publishRate"])
+    values = []
+    with open(f"{output}/{output_name}.log", "w") as output_log:
+        for dataset in datasets:
+            values.append(dataset["aggregatedPublishLatency99pct"])
+            values.append(dataset["aggregatedEndToEndLatency99pct"])
+            output_log.write(dataset["workload"][5:] + "\t" + str(dataset["aggregatedPublishLatency99pct"]) + "\t" + str(dataset["aggregatedEndToEndLatency99pct"]) + "\n")
+    values.sort()
+    low_boundary = values[int(len(values)*0.65)]*1.2
+    high_boundary = values[-1]*1.2
+    
+    with open(f"{output}/{output_name}.gnuplot", "w") as output_gnuplot:
+        output_gnuplot.write(jinja2.Template(BAR).render(
+            output = output_name,
+            title = title,
+            high_boundary = high_boundary,
+            low_boundary = low_boundary,
+            log = f"{output_name}.log"))
+    gnuplot(f"{output_name}.gnuplot", _cwd=output)
+    rm(f"{output}/{output_name}.gnuplot")
+    rm(f"{output}/{output_name}.log")
+
+    
+def report(dir_path, version):
+    trend = defaultdict(lambda: [])
+    for path in os.listdir(dir_path):
+        # check if current path is a file
+        if not os.path.isfile(os.path.join(dir_path, path)):
+            continue
+        if not path.endswith(".json"):
+            continue
+        with open(os.path.join(dir_path, path)) as f:
+            dataset = json.load(f)
+        if dataset["workload"].startswith("load."):
+            trend[dataset["driver"]].append(dataset)
+        rate = dataset["sampleRateMillis"] / 1000
+        analyse(os.path.join(dir_path, "report", dataset["workload"] + "-" + dataset["driver"]), dataset, rate, f"{version} " + dataset["workload"] + " " + dataset["driver"] + " ")
+    for key in trend.keys():
+        if len(trend[key]) == 1:
+            continue
+        bar(os.path.join(dir_path, "report"), trend[key], f"load {key}", f"{version} {key}")
+        
+
+report(argv[1], argv[2])
```

Comment:
I haven't gone thru the whole thing but a README would be nice.. Also we already have generate_charts.py and its wrapper, so its confusing which one to use.. perhaps move to bin/gnuplot or something?